kartothek/io/dask/bag.py

```diff
@@ -207,6 +207,12 @@ def store_bag_as_dataset(
     if not overwrite:
         raise_if_dataset_exists(dataset_uuid=dataset_uuid, store=store)
 
+    partition_secondary_overlap = set(partition_on) & set(secondary_indices)
```

Comment:
Can you turn this into a generic check instead of duplicating it three times?

```python
def check_partition_secondary_overlap(partition_on: Iterable[str], secondary_indices: Iterable[str]) -> None:
    __tracebackhide__ = True

    partition_secondary_overlap = set(partition_on) & set(secondary_indices)
    if partition_secondary_overlap:
        raise RuntimeError(
            f"Cannot create secondary index on partition columns: {partition_secondary_overlap}"
        )
```

Reply:
Unfortunately rest of code is duplicated as well and I’m not sure where to put all that stuff

Reply:
While that's not a _good_ place, the `io_components.utils` module is our bin for collecting this stuff

Reply:
OK, will move it there.